File: 496-NextGreaterElementI/496-NextGreaterElementI.java
// Last updated: 04/04/2026, 23:34:07
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        Map<Integer,Integer> m = new HashMap<>();
        Stack<Integer> st = new Stack<>();
        for(int i=nums2.length-1;i>=0;i--) {
            while((!st.isEmpty()) && nums2[i] >= st.peek()) {
                st.pop();
            }
            if(st.isEmpty()) {
                m.put(nums2[i],-1);
            } else {
                m.put(nums2[i],st.peek());
            }
            st.push(nums2[i]);
        }
        int l = nums1.length;
        int[] ans = new int[l];
        for(int i=0;i<l;i++) {
            ans[i]=m.get(nums1[i]);
        }
        return ans;
    }
}

File: 496-NextGreaterElementI/496-NextGreaterElementI.cpp
// Last updated: 04/04/2026, 23:34:03
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        int len1 = nums1.size();
        int len2 = nums2.size();
        vector<int> ans(len1);
        unordered_map<int,int> ngeMap;
        for(int i=0;i<len2;i++) {
            int key = nums2[i];
            int val = -1;
            for(int j=i+1;j<len2;j++) {
                if(nums2[j] > key) {
                    val = nums2[j];
                    break;
                }
            }
            ngeMap[key] = val;
        }
        int idx = 0;
        for(int num : nums1) {
            ans[idx++] = ngeMap[num];
        }
        return ans;
    }
};